packages/grid/src/vaadin-grid-column-group.js

```diff
@@ -239,6 +250,18 @@ class GridColumnGroup extends ColumnBaseMixin(PolymerElement) {
     }
   }
 
+  /** @private */
+  _groupFrozenToEndChanged(frozenToEnd, rootColumns) {
+    if (rootColumns === undefined || frozenToEnd === undefined) {
+      return;
+    }
+
+    // Don’t propagate the default `false` value.
+    if (frozenToEnd !== false) {
+      Array.from(rootColumns).forEach((col) => (col.frozenToEnd = frozenToEnd));
+    }
```

Comment:
Not sure code is the reason for the issue I found. 

I added one frozen column to a group and both the sibling and parent group got frozen as well. But if I set the `frozenToEnd` property on the column I explicitly set to frozen before, then the other columns don't get the new value propagated to them.

See the recording below:


https://user-images.githubusercontent.com/262432/158830045-035c491d-2bfb-44d6-869a-dc386d8203e5.mp4

Reply:
Looks like this is also a problem with the current implementation: setting column group `frozen` property to `true` gets propagated for `frozen`, but resetting it to `false` doesn't. I will submit a separate PR.

Reply:
Created #3581 where it's explained how the current logic works. It's still a question if we want to let columns be set from `frozen` to `frozenToEnd` directly: I would expect the user to first set `frozen` to false and then reorder.

Reply:
Sorry, I think I have described the issue badly. The column is set to `frozenToEnd` initially and then I set it to `false`. That's when the issue happens, and the new value is not propagated.

Reply:
Just to clarify, is it so that `frozen` works this way but `frozenToEnd` does not?

Reply:
I haven't tested it before with `frozen` only, but testing it now it seems that the same issue is present with `frozen` as well.